code/controllers/subsystem/shuttle.dm

```diff
@@ -217,6 +220,93 @@ SUBSYSTEM_DEF(shuttle)
 	log_game("[key_name(user)] has called the shuttle.")
 	message_admins("[key_name_admin(user)] has called the shuttle.")
 
+	if(legitimacy_check)
+		check_call_legitimacy()
+
+
+/datum/controller/subsystem/shuttle/proc/check_call_legitimacy()
+	if(station_in_trouble())
+		msglog_admins("Shuttle call is legitimate by the listed metrics.")
+		return
+
+	var/time_to_recall = rand(600, 1500)
+	recall_timer_id = addtimer(CALLBACK(src, .proc/recall_illegitimate, emergency.timer), time_to_recall, TIMER_STOPPABLE)
+	if(!annoyed_admiral_message)
+		annoyed_admiral_message = pick("Do you know how expensive these stations are?","Stop wasting my time.","I was sleeping, thanks a lot.","Stand and fight you cowards!","You knew the risks coming in.","Stop being paranoid.","Whatever's broken just build a new one.","No.", "<i>null</i>","<i>Error: No comment given.</i>", "It's a good day to die!")
+
+	message_admins("Shuttle will recall automatically. (<A HREF='?_src_=holder;stop_autorecall=\ref[usr]'>STOP AUTORECALL</A>) (<A HREF='?_src_=holder;set_annoyed_admiral_message=1'>SET ANNOYED ADMIRAL MESSAGE</A>)")
+
+/datum/controller/subsystem/shuttle/proc/recall_illegitimate(old_timer)
+	if(emergency.mode != SHUTTLE_CALL || emergency.timer != old_timer)
+		return
+
+	emergency.cancel(/area/centcom)
+	msglog_admins("Shuttle has been automatically recalled.")
+
+	sleep(50)
+	var/intercepttext = "<FONT size = 3><b>NanoTrasen Update</b>: Request For Shuttle.</FONT><HR>\
+						To whom it may concern:<BR><BR>\
+						We have taken note of the situation upon [station_name()] and have come to the \
+						conclusion that it does not warrant the abandonment of the station.<BR>\
+						If you do not agree with our opinion we suggest that you open a direct \
+						line with us and explain the nature of your crisis.<BR><BR>\
+						<i>This message has been automatically generated based upon readings from long \
+						range diagnostic tools. To assure the quality of your request every finalized report \
+						is reviewed by an on-call rear admiral.<BR>\
+						<b>Rear Admiral's Notes:</b> \
+						[annoyed_admiral_message]"
+	annoyed_admiral_message = null
+	print_command_report(intercepttext, announce=TRUE)
+
+/datum/controller/subsystem/shuttle/proc/station_in_trouble() //Checks are sorted in rough order of processing cost
+	. = FALSE
+	msglog_admins("Emergency shuttle legitimacy checks follow:")
+
+	listclearnulls(SSshuttle.horrible_things)
+	var/num_things = SSshuttle.horrible_things.len
+	msglog_admins("[num_things] horrible things on station[num_things ? ":[english_list(SSshuttle.horrible_things)]" : ""]")
```

Comment:
If there are horrible things on station, they'd be things like "nuclear ops on station", "dominator", "Ratvar", "doomsday device" that sort of thing.